auth/store.go

```diff
@@ -150,6 +150,9 @@ type AuthStore interface {
 
 	// CheckPassword checks a given pair of username and password is correct
 	CheckPassword(username, password string) (uint64, error)
+
+	// IsAuthEnabled checks authentication is enabled or not
+	IsAuthEnabled() bool
```

Comment:
`IsEnabled` stuff leads to a lot of split paths. Ideally it shouldn't be necesary; if auth is disabled either the authstore would be niled out (still has path splitting, though) or the authstore would use a dummy auth that passes everything through.